biodata-models/src/main/java/org/opencb/biodata/models/variant/Variant.java

```diff
@@ -219,7 +217,9 @@ public String getReference() {
 
     public void setReference(String reference) {
         this.reference = reference;
-        this.length = Math.max(reference.length(), alternate.length());
+        if (reference != null && alternate != null) {
+            this.length = Math.max(reference.length(), alternate.length());
```

Comment:
If alternate, reference, or length changes the type should be changed effectively, I really hate that this class as it is now requires to recalculate the type after every change. That parameter should be directly calculated to avoid the need to be aware of that circumstance in every modification